ibm_db.c

```diff
@@ -17174,10 +17071,15 @@ static PyObject *ibm_db_fetchall(PyObject *self, PyObject *args)
                 Py_XDECREF(result_list);
                 return NULL;
             }
+            else
+            {
+                Py_XDECREF(return_value);
+            }
         }
         else
         {
             LogMsg(DEBUG, "Fetched value is not a valid row, breaking loop", fileName);
+            Py_XDECREF(return_value);
             break;
         }
```

Comment:
Call `Py_XDECREF(return_value);` here after else block and just before end of while loop. Rest other calls of `Py_XDECREF(return_value);` inside while loop not required.